crates/uv-resolver/src/lock/mod.rs

```diff
@@ -3883,6 +3883,48 @@ impl TryFrom<SourceWire> for Source {
     }
 }
 
+/// Convert a path to a relative path, unless the original user input was an absolute path
+/// and the relative path would require traversing outside the root directory (i.e., starts with `..`).
+///
+/// This preserves user intent: if a user specifies an absolute path (e.g., `/shared/wheels`),
+/// we keep it absolute in the lock file rather than converting to a relative path like
+/// `../../shared/wheels` which may break when used from different directory depths.
+fn relative_to_or_absolute(
+    path: &Path,
+    root: &Path,
+    given: Option<&str>,
+) -> Result<PathBuf, io::Error> {
+    // Check if the original user input was an absolute path.
+    // We need to distinguish between:
+    // 1. User explicitly wrote an absolute path (e.g., `/shared/wheels` in pyproject.toml)
+    // 2. System resolved a relative path to absolute (e.g., workspace member `../child` -> `/tmp/.../child`)
+    //
+    // For case 1, `given` would be the absolute path itself or a file:// URL pointing to it.
+    // For case 2, `given` would be the original relative path (e.g., `../child`).
+    //
+    // We only preserve absolute paths for case 1, where the user explicitly specified an absolute path.
+    let is_user_specified_absolute = given.is_some_and(|g| {
+        let given_path = g.strip_prefix("file://").unwrap_or(g);
+        Path::new(given_path).is_absolute()
+            && path
+                .to_str()
+                .is_some_and(|p| p == given_path || given_path.ends_with(p))
+    });
+
+    match relative_to(path, root) {
+        Ok(relative) => {
+            // If the user explicitly specified an absolute path and the relative path starts with "..",
+            // preserve the absolute path to maintain portability across directory depths.
+            if is_user_specified_absolute && relative.starts_with("..") {
+                std::path::absolute(path)
```

Comment:
I'm not following here, if the user input was absolute, why don't we preserve it?